core/db/db.go

```diff
@@ -36,6 +36,8 @@ type Event interface {
 	InstanceID() InstanceID
 	// Collection is the associated instance's collection name.
 	Collection() string
+	// Marshal the event to JSON.
+	Marshal() ([]byte, error)
```

Comment:
This is used to marshal the event to something that can be leveraged in a JS validator function.